modules/signals/src/state-source.ts

```diff
@@ -49,6 +54,10 @@ export function patchState<State extends object>(
   notifyWatchers(stateSource);
 }
 
+function freezeInDevMode<State extends object>(value: State): State {
+  return ngDevMode ? deepFreeze(value) : value;
```

Comment:
I think we should add this to the v19 list as a breaking change.
I suppose this could break some applications in dev mode.

Reply:
Fair enough. It really throws an error and I couldn't find a way how to "map" it to a warning message.